### PR TITLE
Mldb 1841 distinct on

### PR DESCRIPTION
**Pull request by steadtler**

### CODE REVIEW

**Comment by steadtler** [comment]
working on the doc

File: container_files/public_html/doc/builtin/sql/SelectExpression.md [hunk 1]
```diff
@@ -92,6 +92,25 @@ The following functions are available in the context of a column expression:
   of elements in the column path, NULL will be returned.
 - `rowCount()` is the number of rows that have a value for this column, including explicit NULLs.
 
+## Filtering duplicated rows based on an expression
+
+It is possible to filter out rows based on the value of an expression using the `DISTINCT` ON optional
```
Comment (guyd) [inline container_files/public_html/doc/builtin/sql/SelectExpression.md]: wrap ON with the backticks here also

File: container_files/public_html/doc/builtin/sql/SelectExpression.md [hunk 2]
```diff
@@ -92,6 +92,25 @@ The following functions are available in the context of a column expression:
...
+clause. The syntax is as follow:
```
Comment (guyd) [inline container_files/public_html/doc/builtin/sql/SelectExpression.md]: as follow*s*?

File: testing/MLDB-1841-distinct-on.py [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+# Mathieu Marquis Bolduc, 2016-07-21
+# This file is part of MLDB. Copyright 2016 Datacratic. All rights reserved.
+#
+
+import unittest
+import json
+
+mldb = mldb_wrapper.wrap(mldb) # noqa
+
+#### TODO: CHECK LIMIT / OFFSET
```
Comment (guyd) [inline testing/MLDB-1841-distinct-on.py]: This has been covered I believe.

**Comment by guyd** [comment]
Very good test coverage.
reviewed briefly the c++ code
+1

**Pull request merged**